### ISSUE TITLE
AI is not defined

### ISSUE THREAD

**Issue by jacobslusser** [TrilonIO/angular-application-insights#65]
When trying to set the severity level of a call to `trackTrace`, I get the following runtime error:

```
ReferenceError: AI is not defined
```

The call looks like this:

```ts
this.applicationInsights.trackTrace(message, properties, AI.SeverityLevel.Information);
```

I've tried several imports to resolve this reference but can't seem to find the right one. Any thoughts?

**Comment by MarkPieszak** [comment]
AI should be automatically added to your `@types/` folder in node_modules after installing, it comes from **`applicationinsights-js`** itself. Can you double check if it's there?

Was the package installed correctly? I'd wipe out node_modules and do another `npm i`. Make sure `"@markpieszak/ng-application-insights": "^7.0.1",` is in your package.json as well.

![image](https://user-images.githubusercontent.com/2574412/48854994-34f78180-ed81-11e8-88f5-d8c781885cb0.png)

**Comment by jacobslusser** [comment]
Yes, the dependency was brought in and autocomplete in vscode finds it, however, if I'm meant to add an `import` statement for `AI` then I'm not sure how to do it. No syntax errors, but runtime error is what I posted above.

I'm using the latest version of `@markpieszak/ng-application-insights`.

**Comment by nachiroy** [comment]
I'm facing this same issue. I checked that the correct '@types' are present in the typeRoots of my root tsconfig.json. Is there some step that we are missing?

**Comment by MarkPieszak** [comment]
Are you guys still seeing this one? I'm strangely unable to reproduce this one.. :(

**Comment by jacobslusser** [comment]
No, I was never able to get it working as expected. As a workaround I just copied in the `SeverityLevel` source into my file like this:

```ts
import { Injectable } from '@angular/core';
import { AppInsightsService } from '@markpieszak/ng-application-insights';
import { environment } from '../../../environments/environment';

enum SeverityLevel {
  Verbose = 0,
  Information = 1,
  Warning = 2,
  Error = 3,
  Critical = 4,
}

@Injectable({
  providedIn: 'root'
})
export class LoggerService {

  constructor(private applicationInsights: AppInsightsService) {
  }

  /**
   * Writes a verbose log message.
   * This should be called for information that's typically valuable only for debugging.
   * @param message The message to log.
   * @param properties An optional string-string dictionary of additional properties to include.
   */
  logVerbose(message: string, properties?: { [name: string]: string }) {
    if (message) {
      this.applicationInsights.trackTrace(message, properties, SeverityLevel.Verbose);
      if (!environment.production && console.debug) {
        console.info(`VERBOSE: ${message}`, properties);
      }
    }
  }
}
```

**Comment by richardpineo** [comment]
We have seen this error in our applications as well. Unfortunately I don't have a good repo to reproduce but we are using the same workaround that @jacobslusser suggested above.

**Comment by MarkPieszak** [comment]
So to go along with @jacobslusser's custom enum for SeverityLevel, I created one and exported it inside the package here.

I also updated the Type defs to except both SeverityLevel (I provide in there), and AI.SeverityLevel just incase either work for people!

Hopefully this helps make life a little bit easier - apologies on the delay. Life has been hectic getting **[Trilon.io](https://trilon.io)** all setup!

7.0.2 pushed to NPM with these updates :)
